src/Microsoft.AspNetCore.SignalR.Client/ConnectionFactory.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO.Pipelines;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    // REVIEW: Split into Microsoft.AspNetCore.Sockets.Client?
+    public class ConnectionFactory
+    {
+        private readonly HttpClient _httpClient;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly PipelineFactory _pipelineFactory;
+        private readonly ILogger _logger;
+
+        public ConnectionFactory(HttpClient httpClient, PipelineFactory pipelineFactory, ILoggerFactory loggerFactory)
```

Comment:
What's the point of taking PipelineFactory? Does not look like you can provide your own useful implementation - there is no interface and no virtual methods. 
@davidfowl - this PipeLineFactory makes code hard to test - should it be changed or we need to provide our own interface delegating the code to the default implementation?